sdk/go/common/testing/environment.go

```diff
@@ -76,11 +76,15 @@ func WriteYarnRCForTest(root string) error {
 
 // NewEnvironment returns a new Environment object, located in a temp directory.
 func NewEnvironment(t *testing.T) *Environment {
-	root := t.TempDir()
+	//nolint:usetesting // We control the lifecycle of the environment.
+	root, err := os.MkdirTemp("", "test-env")
+	assert.NoError(t, err, "creating temp directory")
 	assert.NoError(t, WriteYarnRCForTest(root), "writing .yarnrc file")
 
 	// We always use a clean PULUMI_HOME for each environment to avoid any potential conflicts with plugins or config.
-	home := t.TempDir()
+	//nolint:usetesting // We control the lifecycle of the environment.
+	home, err := os.MkdirTemp("", "test-env-home")
```

Comment:
One thing I noticed while here: Should we be cleaning up this directory? This was originally introduced in https://github.com/pulumi/pulumi/pull/15559, but we don't delete it in `DeleteEnvironment`.

Reply:
Yeh we should be

Reply:
Added a commit that does this